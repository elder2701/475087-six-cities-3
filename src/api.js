import axios from "axios";

const Error = {

};

export const createAPI = (onUnauthorized) => {
  const api = axios.create({
    baseURL: `https://htmlacademy-react-3.appspot.com/six-cities`,
    timeount: 5000,
    withCredentials: true
  });

  const onSuccess = (response) => {
    return response;
  };

  const onFail = (err) => {
    const {response} = err;

    if (response.status === Error.UNAUTHORIZED) {
      onUnauthorized();
      throw err;
    }
    throw err;
  };
  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
