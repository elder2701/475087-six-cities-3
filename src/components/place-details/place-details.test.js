import React from "react";
import renderer from "react-test-renderer";
import PlaceDetails from "./place-details.jsx";
import createMapBlock from "../map/create-map-block.js";
import cofigureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducer/name-space.js";
import thunk from "redux-thunk";
import {createAPI} from "../../api.js";

const api = createAPI(()=>{});

const mockStore = cofigureStore([thunk.withExtraArgument(api)]);

it(`<PlaceDetails /> sould be render`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      offers: {
        Amsterdam: {
          city: {
            name: `Amsterdam`,
            location: {
              latitude: 50.846557,
              longitude: 4.351697,
              zoom: 13
            }
          },
          offers: [
            {
              images: [],
              title: `The house among olive `,
              rating: 3.4,
              type: `apartment`,
              bedrooms: 3,
              price: 148,
              goods: [
                `Washer`,
                `Breakfast`,
                `Laptop friendly workspace`,
                `Dishwasher`,
                `Towels`,
                `Baby seat`,
                `Fridge`,
                `Air conditioning`
              ],
              description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
              location: {
                latitude: 50.869557,
                longitude: 4.332697,
                zoom: 16
              },
              id: 1,
              previewImage: ``,
              isFavorite: false,
              isPremium: true,
              maxAdults: 4,
              hostId: 25,
              hostName: `Angelina`,
              hostIsPro: true,
              hostAvatarUrl: `img/avatar-angelina.jpg`
            }
          ]
        }
      }
    },
    [NameSpace.CITY]: {
      city: `Amsterdam`
    },
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`,
      userInfo: {
        "id": 1,
        "email": `O@gmail.com`,
        "name": `O`,
        "avatar_url": `/static/avatar/2.jpg`,
        "is_pro": false
      }
    },
    [NameSpace.OFFER]: {
      id: `1`,
      offerAround: [],
      comments: [],
      isSendComment: true
    }
  });
  createMapBlock();
  const tree = renderer
    .create(
        <Provider store={store}>
          <PlaceDetails idOffer={`1`} updateStatus={()=>{}} selectOffer={()=>{}}/>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
