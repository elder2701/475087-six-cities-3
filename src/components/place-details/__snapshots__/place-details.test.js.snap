// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceDetails /> sould be render 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      />
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            The house among olive 
          </h1>
          <button
            className="property__bookmark-button button "
            onClick={[Function]}
            type="button"
          >
            <svg
              className="property__bookmark-icon place-card__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            3.4
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            apartment
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            3
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            4
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            148
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Washer
            </li>
            <li
              className="property__inside-item"
            >
              Breakfast
            </li>
            <li
              className="property__inside-item"
            >
              Laptop friendly workspace
            </li>
            <li
              className="property__inside-item"
            >
              Dishwasher
            </li>
            <li
              className="property__inside-item"
            >
              Towels
            </li>
            <li
              className="property__inside-item"
            >
              Baby seat
            </li>
            <li
              className="property__inside-item"
            >
              Fridge
            </li>
            <li
              className="property__inside-item"
            >
              Air conditioning
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="/img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews ·
            <span
              className="reviews__amount"
            >
              0
            </span>
          </h2>
          <ul
            className="reviews__list"
          />
          <form
            action="#"
            className="reviews__form form"
            method="post"
            onSubmit={[Function]}
          >
            <label
              className="reviews__label form__label"
              htmlFor="review"
            >
              Your review
            </label>
            <div
              className="reviews__rating-form form__rating"
            >
              <input
                checked=""
                className="form__rating-input visually-hidden"
                disabled=""
                id="5-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={5}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="5-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked=""
                className="form__rating-input visually-hidden"
                disabled=""
                id="4-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={4}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="4-stars"
                title="good"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked=""
                className="form__rating-input visually-hidden"
                disabled=""
                id="3-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={3}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="3-stars"
                title="not bad"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked=""
                className="form__rating-input visually-hidden"
                disabled=""
                id="2-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={2}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="2-stars"
                title="badly"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked=""
                className="form__rating-input visually-hidden"
                disabled=""
                id="1-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={1}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="1-stars"
                title="terribly"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
            </div>
            <textarea
              className="reviews__textarea form__textarea"
              disabled=""
              id="review"
              name="review"
              onChange={[Function]}
              placeholder="Tell how was your stay, what you like and what can be improved"
              value=""
            />
            <div
              className="reviews__button-wrapper"
            >
              <p
                className="reviews__help"
              >
                To submit review please make sure to set
                <span
                  className="reviews__star"
                >
                  rating
                </span>
                 and describe your stay with at least
                <b
                  className="reviews__text-amount"
                >
                  50 characters
                </b>
                .
              </p>
              <button
                className="reviews__submit form__submit button"
                disabled="disabled"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
    <section
      className="property__map map"
      id="map"
    />
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      />
    </section>
  </div>
</main>
`;
